src/app/edit-product/edit-product.component.ts

```diff
@@ -0,0 +1,70 @@
+import {Component, OnInit} from '@angular/core';
+import {FormBuilder, Validators} from "@angular/forms";
+import {ProductService} from "../services/product.service";
+import {ProductItemDetailed} from "../interfaces/ProductItemDetailed";
+import {ActivatedRoute} from "@angular/router";
+
+@Component({
+  selector: 'app-edit-product',
+  templateUrl: './edit-product.component.html',
+  styleUrls: ['./edit-product.component.scss']
+})
+
+export class EditProductComponent implements OnInit {
+  // editForm = new FormGroup({
+  //   name: new FormControl(''),
+  //   price: new FormControl(-1),
+  // });
+
+  editForm = this.formBuilder.group({
+    name: ['', Validators.required],
+    category: ['', Validators.required],
+    price: [-1, [Validators.required, Validators.min(1)]],
+    description: ['', Validators.required],
+    image: ['', Validators.required]
+  })
+
+  product: ProductItemDetailed = {
+    id: -1,
+    name: "name",
+    category: "category",
+    price: -1,
+    description: "description",
+    image: "image"
+  };
+
+  constructor(private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private productService: ProductService) {
+  }
+
+  ngOnInit(): void {
+    const id = Number(this.route.snapshot.paramMap.get('id'));
+    this.productService.getProduct(id).subscribe(product => {
```

Comment:
**suggestion:** take care to avoid memory leaks in your code by unsubscribing from observables. https://www.thisdot.co/blog/best-practices-for-managing-rxjs-subscriptions/